common/faults/faults.c

```diff
@@ -0,0 +1,461 @@
+/**
+ * @file faults.c
+ * @author Aditya Anand (anand89@purdue.edu)
+ * @brief Creating a library of faults to create an easy to debug atmosphere on the car
+ * @version 0.1
+ * @date 2022-05-11
+ *
+ * @copyright Copyright (c) 2022
+ *
+ */
+#include "faults.h"
+
+
+//BEGIN AUTO INCLUDES
+#if FAULT_NODE_NAME == 0
+	#include "source/main_module/can/can_parse.h"
+#endif
+#if FAULT_NODE_NAME == 1
+	#include "source/driveline/can/can_parse.h"
+#endif
+#if FAULT_NODE_NAME == 2
+	#include "source/dashboard/can/can_parse.h"
+#endif
+#if FAULT_NODE_NAME == 3
+	#include "source/precharge/can/can_parse.h"
+#endif
+#if FAULT_NODE_NAME == 4
+	#include "source/torque_vector/can/can_parse.h"
+#endif
+#if FAULT_NODE_NAME == 5
+	#include "source/l4_testing/can/can_parse.h"
+#endif
+//END AUTO INCLUDES
+
+//BEGIN AUTO FAULT INFO ARRAY DEFS
+int idArray[TOTAL_NUM_FAULTS] = { ID_BATT_FLOW_FAULT, ID_DRIVE_FLOW_FAULT, ID_MAIN_COMM_FAULT, ID_LV_DEAD_FAULT, ID_MOT_FRONT_OT_FAULT, ID_WLSPD_L_FAULT,
+			ID_WLSPD_R_FAULT, ID_DRIVELINE_COMM_FAULT, ID_BMS_FAULT, ID_BATT_OT_FAULT, ID_TV_OFFLINE_FAULT, ID_TEST_FAULT_1_FAULT, ID_TEST_FAULT_2_FAULT,
+			ID_TEST_FAULT_3_FAULT, ID_TEST_FAULT_4_FAULT,};
+int maxArray[TOTAL_NUM_FAULTS] = { BATT_FLOW_MAX, DRIVE_FLOW_MAX, MAIN_COMM_MAX, LV_DEAD_MAX, MOT_FRONT_OT_MAX, WLSPD_L_MAX,
+			WLSPD_R_MAX, DRIVELINE_COMM_MAX, BMS_MAX, BATT_OT_MAX, TV_OFFLINE_MAX, TEST_FAULT_1_MAX, TEST_FAULT_2_MAX,
+			TEST_FAULT_3_MAX, TEST_FAULT_4_MAX,};
+int minArray[TOTAL_NUM_FAULTS] = { BATT_FLOW_MIN, DRIVE_FLOW_MIN, MAIN_COMM_MIN, LV_DEAD_MIN, MOT_FRONT_OT_MIN, WLSPD_L_MIN,
+			WLSPD_R_MIN, DRIVELINE_COMM_MIN, BMS_MIN, BATT_OT_MIN, TV_OFFLINE_MIN, TEST_FAULT_1_MIN, TEST_FAULT_2_MIN,
+			TEST_FAULT_3_MIN, TEST_FAULT_4_MIN,};
+fault_priority_t priorityArray[TOTAL_NUM_FAULTS] = { BATT_FLOW_PRIORITY, DRIVE_FLOW_PRIORITY, MAIN_COMM_PRIORITY, LV_DEAD_PRIORITY, MOT_FRONT_OT_PRIORITY, WLSPD_L_PRIORITY,
+			WLSPD_R_PRIORITY, DRIVELINE_COMM_PRIORITY, BMS_PRIORITY, BATT_OT_PRIORITY, TV_OFFLINE_PRIORITY, TEST_FAULT_1_PRIORITY, TEST_FAULT_2_PRIORITY,
+			TEST_FAULT_3_PRIORITY, TEST_FAULT_4_PRIORITY,};
+char msgArray[TOTAL_NUM_FAULTS][MAX_MSG_SIZE] = { BATT_FLOW_MSG, DRIVE_FLOW_MSG, MAIN_COMM_MSG, LV_DEAD_MSG, MOT_FRONT_OT_MSG, WLSPD_L_MSG,
```

Comment:
Mabye, but likely only with the msgArray. A lot of my fault library, especially for a board as critical as main, would need a lot of the other values since it would need to determine first if any faults are latched, and next the priority of the fault, since it would either do nothing, exit ready2drive, or disable hv